src/main/java/com/iotiq/application/domain/Cart.java

```diff
@@ -0,0 +1,39 @@
+package com.iotiq.application.domain;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.data.jpa.domain.AbstractPersistable;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Entity
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class Cart extends AbstractPersistable<UUID> {
+
+    @OneToOne(cascade = CascadeType.ALL)
```

Comment:
Please refer to: https://vladmihalcea.com/the-best-way-to-map-a-onetoone-relationship-with-jpa-and-hibernate/
for OneToOne relations. Cart is not the parent entity here. Actually customer is. We should cascade actions starting from Customer.